Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -0,0 +1,619 @@
+        }
+
+        var keyView = new DataView(keyBuffer);
+
+        var dp = 0;
+        var dpend = data.byteLength;
+        var dpend64 = dpend - (dpend % 8);
+        var kpend = keylen;
+        var kp;
+        var off = 8;
+
+        // This algorithm is intentionally asymmetric to make it more difficult to
+        // guess. Security through obscurity. :-(
+
+        // while we have a full uint64 (8 bytes) left to do
+        // assumes buffer is 64bit aligned (or processor doesn't care)
+        while (dp < dpend64) {
+            // rotate the key each time through by using the offets 16,0,8,16,0,8,...
+            off = (off + 8) % 24;
+            kp = off;
+
+            // run through one key length xor'ing one uint64 at a time
+            // then drop out to rotate the key for the next bit
+            while ((dp < dpend64) && (kp < kpend)) {
+                dataView.setUint32(dp, dataView.getUint32(dp, true) ^ keyView.getUint32(kp, true), true);
+                dataView.setUint32(dp+4, dataView.getUint32(dp+4, true) ^ keyView.getUint32(kp+4, true), true);
+                dp += 8;
+                kp += 24;
+            }
+        }
+
+        // now the remaining 1 to 7 bytes
+        if (dp < dpend) {
+            if (kp >= kpend) {
+                // rotate the key one last time (if necessary)
+                off = (off + 8) % 24;
+                kp = off;
+            }
+
+            while (dp < dpend) {
+                dataView.setUint8(dp, dataView.getUint8(dp) ^ keyView.getUint8(kp));
+                dp++;
+                kp++;
+            }
+        }
+    };
+
+
+    var qtMagic = 32301;
+    var compressedMagic = 0x7468dead;
+    var compressedMagicSwap = 0xadde6874;
+    /**
+     * Uncompresses a Google Earth Enterprise packet.
+     *
+     * @param {ArrayBuffer} data The data to be uncompressed.
+     *
+     * @private
+     */
+    GoogleEarthEnterpriseMetadata.uncompressPacket = function (data) {
+        // The layout of this decoded data is
+        // Magic Uint32
+        // Size Uint32
+        // [GZipped chunk of Size bytes]
+
+        // Pullout magic and verify we have the correct data
+        var dv = new DataView(data);
+        var offset = 0;
+        var magic = dv.getUint32(offset, true);
+        offset += sizeOfUint32;
+        if (magic !== compressedMagic && magic !== compressedMagicSwap) {
+            throw new RuntimeError('Invalid magic');
+        }
+
+        // Get the size of the compressed buffer
+        var size = dv.getUint32(offset, true);
+        offset += sizeOfUint32;
+        if (magic === compressedMagicSwap) {
+            size = ((size >>> 24) & 0x000000ff) |
+                    ((size >>>  8) & 0x0000ff00) |
+                    ((size <<  8) & 0x00ff0000) |
+                    ((size << 24) & 0xff000000);
+        }
+
+        var compressedPacket = new Uint8Array(data, offset);
+        var uncompressedPacket = pako.inflate(compressedPacket);
```

Comment:
It's probably easy, not sure if it will get us that much. Let me get the performance numbers vs. quantized mesh to see if we really need to investigate performance.